### DISCUSSION TITLE
Gallery for extension

### DISCUSSION THREAD

**Discussion by CarsonSlovoka** [yuin/goldmark#325] [Ideas]
This is a great library👍 and I noticed that it contains some useful [extensions included](https://github.com/yuin/goldmark/tree/20df169/extension), [build-in extensions](https://github.com/yuin/goldmark#built-in-extensions)
I want to ask questions like  
- Math{Katex, Latex}
  - [issues/32](https://github.com/yuin/goldmark/issues/32)
  - [discussions/322](https://github.com/yuin/goldmark/discussions/322)
- [issues/56](https://github.com/yuin/goldmark/issues/56)<sup>unicode heading IDs</sup>
- [Allow mixed case attribute names](https://github.com/yuin/goldmark/discussions/279) <sup>`text {linenos=true linenostart=42}`</sup>

Do they have any chance to be included in this extension (build-in extensions)?
Or **what are the requirements satisfied to be included in the extension?** (100% [commonmark-spec](https://github.com/commonmark/)?)

----

If the criteria for being built-extension are very stringent, can we **provide a gallery-like area** where people can share their extension?
For example, create a new discussion for that (and add a table on the README?). Then, user can share their links, and people can evaluate them to increase credibility and give newcomers<sup>[dis-212](https://github.com/yuin/goldmark/discussions/212), [is-163](https://github.com/yuin/goldmark/issues/163)</sup> directions on how to make extensions.

**Comment by yuin** [comment]
> Do they have any chance to be included in this extension (build-in extensions)?
Or what are the requirements satisfied to be included in the extension? (100% [commonmark-spec](https://github.com/commonmark/)?)

Basic policy: Extensions can be implemented as an external library should be implemented as an external library to keep goldmark simple for maintainability.

Why goldmark has built-in extensions?

1. Historical reason: blackfriday was a de facto-standard library. Many users will be migrate from blackfriday to goldmark. So I added extensions that are included in blackfriday as a built-in extensions.
2. It can not be implemented as an external library

> If the criteria for being built-extension are very stringent, can we provide a gallery-like area where people can share their extension?

Just create a PR that adds an extension to the README.

**Reply by CarsonSlovoka** [reply]
I did not notice that the original README file included the [plugin list](https://github.com/yuin/goldmark/blob/c71a97b8372876d63528b54cedecf1104530fe3b/README.md?plain=1#L433-L447). 😅
